src/core/service.c

```diff
@@ -4140,43 +4197,73 @@ static void service_notify_message(
                 }
         }
 
-        /* Interpret READY=/STOPPING=/RELOADING=. Last one wins. */
-        STRV_FOREACH_BACKWARDS(i, tags) {
+        /* Parse MONOTONIC_USEC= */
+        e = strv_find_startswith(tags, "MONOTONIC_USEC=");
```

Comment:
Rather than relying on the client to set this, wouldn't it be safer to get it from the dbus message via `sd_bus_message_get_monotonic_usec`? That way we are sure nothing can play tricks with us, and there's one fewer things to implement for clients so it's simpler and there's less space for mistakes

Reply:
there is not dbus message involved, and the sd_bus_message_get_monotonic_usec() thing is not really useful. was a kdbus thing, and is not implementable without it currently, as sockets can do CLOCK_REALTIME timestamping, but not CLOCK_MONOTONIC.

I so wished kernel could do CLOCK_MONOTONIC timestamping for sockets. then yes, we could totally drop this. But without it there's just no way…

Reply:
ah, never mind then